File: FFCG.G4.Text/FFCG.G4.Text.Manipulator/ManipulateText/TextManipulator.cs
using System.Collections.Generic;
using System.Linq;

namespace FFCG.G4.Text.Manipulator.ManipulateText
{
    public static class TextManipulator
    {
        public static string Manipulate(this string text, List<IManipulateTextRule> rules)
        {
            foreach (var rule in rules)
                text = rule.Manipulate(text);

            return text;
        }
    }
}

File: FFCG.G5.JSWordManipulator/src/textManipulator.js
class TextManipulator {
    constructor(manipulatorRules) {
        this.manipulatorRules = manipulatorRules
    }

    manipulateText(text) {
        let manipulatedText = text

        this.manipulatorRules.forEach(rule => {
            manipulatedText = rule.manipulate(manipulatedText)
        })

        return manipulatedText
    }
}